superset/assets/javascripts/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -58,22 +61,52 @@ class SqlEditorLeftBar extends React.PureComponent {
   resetState() {
     this.props.actions.resetState();
   }
-  fetchTables(dbId, schema) {
-    const actualDbId = dbId || this.props.queryEditor.dbId;
-    if (actualDbId) {
-      const actualSchema = schema || this.props.queryEditor.schema;
-      this.setState({ tableLoading: true });
-      this.setState({ tableOptions: [] });
-      const url = `/superset/tables/${actualDbId}/${actualSchema}`;
+  getTableNamesBySubStr(input) {
+    if (!this.props.queryEditor.dbId || !input) {
+      return Promise.resolve({ options: [] });
+    }
+    return fetch(
+          `/superset/tables/${this.props.queryEditor.dbId}/`
+          `${this.props.queryEditor.schema}/${input}`,
+          { method: 'GET', credentials: 'include' }
+        )
+        .then(response => response.json())
+		.then(json => ({ options: json.options }));
+  }
+  fetchTables(dbId, schema, substr) {
```

Comment:
it would require me to refactor the existing flow, what is not desired for this PR.
I left a TODO to do that separately.